cpp/include/cugraph_c/properties.h

```diff
@@ -0,0 +1,146 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+//
+// Speculative description of handling generic vertex and edge properties.
+//
+// If we have vertex properties and edge properties that we want to apply to an existing graph
+// (after it was created) we could use these methods to construct C++ objects to represent these
+// properties.
+//
+// These assume the use of external vertex ids and external edge ids as the mechanism for
+// correlating a property to a particular vertex or edge.
+//
+
+#include <cugraph_c/resource_handle.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct {
+  int32_t align_;
+} cugraph_vertex_property_t;
+
+typedef struct {
+  int32_t align_;
+} cugraph_edge_property_t;
+
+typedef struct {
+  int32_t align_;
+} cugraph_vertex_property_view_t;
+
+typedef struct {
+  int32_t align_;
+} cugraph_edge_property_view_t;
+
+#if 0
+// Blocking out definition of these since this is speculative work.
+
+/**
+ * @brief     Create a vertex property
+ *
+ * @param [in]  handle      Handle for accessing resources
+ * @param [in]  graph       Pointer to graph.
+ * @param [in]  vertex_ids  Device array of vertex ids
+ * @param [in]  property    Device array of vertex property
+ * @param [out] result      Pointer to the location to store the pointer to the vertex property object
+ * @param [out] error       Pointer to an error object storing details of any error.  Will
+ *                          be populated if error code is not CUGRAPH_SUCCESS
+ * @return error code
+ */
+cugraph_error_code_t cugraph_vertex_property_create(
+  const cugraph_resource_handle_t* handle,
+  const cugraph_graph_t * graph,
+  const cugraph_type_erased_device_array_t* vertex_ids,
+  const cugraph_type_erased_device_array_t* properties,
+  cugraph_vertex_property_t** result,
+  cugraph_error_t** error);
+
+/**
+ * @brief     Create a edge property
+ *
+ * @param [in]  handle           Handle for accessing resources
+ * @param [in]  graph            Pointer to graph.
+ * @param [in]  lookup_container Lookup map
+ * @param [in]  edge_ids         Device array of edge ids
+ * @param [in]  property         Device array of edge property
+ * @param [out] result           Pointer to the location to store the pointer to the edge property object
+ * @param [out] error            Pointer to an error object storing details of any error.  Will
+ *                               be populated if error code is not CUGRAPH_SUCCESS
+ * @return error code
+ */
+cugraph_error_code_t cugraph_edge_property_create(
+  const cugraph_resource_handle_t* handle,
+  const cugraph_graph_t * graph,
+  const cugraph_lookup_container_t* lookup_container,
+  const cugraph_type_erased_device_array_t* edge_ids,
+  const cugraph_type_erased_device_array_t* properties,
```

Comment:
What happens if edge IDs are provided only for a subset of edges? Will those edges have undefined values? Should we ask for default values?

And no need for an update function? (e.g. update bias values only for a small subset of edges).

Reply:
Reasonable ideas.  I'll look at defining update functions and a mechanism for a default value.  I'll add it to both vertex and edge functions.

Reply:
Just fyi. In the C++ primitives level, we have `fill_edge_property_value` (which sets edge property values for the entire set of edges, we may add a function that works on a subset of the edges in the future) and `update_edge_property` function that works either on a subset of the edges or the entire set of edges. Following this may make implementing the API easier.

Reply:
I pushed an update that adds these functions.  Plan is to implement them at some point in the future.